nodebuilder/gateway/config.go

```diff
@@ -22,10 +23,13 @@ func DefaultConfig() Config {
 }
 
 func (cfg *Config) Validate() error {
-	if ip := net.ParseIP(cfg.Address); ip == nil {
-		return fmt.Errorf("gateway: invalid listen address format: %s", cfg.Address)
+	sanitizedAddress, err := utils.ValidateAddr(cfg.Address)
+	if err != nil {
+		return fmt.Errorf("gateway: invalid address: %w", err)
 	}
-	_, err := strconv.Atoi(cfg.Port)
+	cfg.Address = sanitizedAddress
+
+	_, err = strconv.Atoi(cfg.Port)
```

Comment:
nit: is there a reason this wouldn't be part of `ValidateAddr`?

Reply:
Ref #1385